document/metadata.go

```diff
@@ -0,0 +1,101 @@
+package document
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/loader"
+)
+
+// ParseRegoWithAnnotations parse the rego in the indicated directory
+func ParseRegoWithAnnotations(directory string) (ast.FlatAnnotationsRefSet, error) {
+	// Recursively find all rego files (ignoring test files), starting at the given directory.
+	result, err := loader.NewFileLoader().
+		WithProcessAnnotation(true).
+		Filtered([]string{directory}, func(_ string, info os.FileInfo, _ int) bool {
+			if strings.HasSuffix(info.Name(), "_test.rego") {
+				return true
+			}
+
+			if !info.IsDir() && filepath.Ext(info.Name()) != ".rego" {
+				return true
+			}
+
+			return false
+		})
+
+	if err != nil {
+		return nil, fmt.Errorf("filter rego files: %w", err)
+	}
+
+	if _, err := result.Compiler(); err != nil {
+		return nil, fmt.Errorf("compile: %w", err)
+	}
+
+	compiler := ast.NewCompiler()
+	compiler.Compile(result.ParsedModules())
+	as := compiler.GetAnnotationSet().Flatten()
+
+	return as, nil
+}
+
+type Section struct {
+	H           string
+	Path        string
+	Annotations *ast.Annotations
+}
+
+func (s Section) Equal(s2 Section) bool {
```

Comment:
I guess this is not being used, is it?

Reply:
It is used in the tests; I will add a comment to explain. In short, `cmp.Equal` will use the `Equal` method of a struct to assess equality instead of deep equality. The `*ast.Annotations` are too complex, and we don't want to have to validate all the fields to infer equality